src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,247 @@
+            if "PyQt5" in mod:
+                try:
+                    from PyQt5.QtCore import QT_VERSION_STR, PYQT_VERSION_STR
+                except:
+                    unattainable_modules.append(["Qt", "PyQt"])
+                else:
+                    module_versions_dict["Qt"] = QT_VERSION_STR
+                    module_versions_dict["PyQt"] = PYQT_VERSION_STR
+                continue
+
+            # Different attempts of retrieving the modules version
+            try:
+                module_versions_dict[mod] = __import__(mod).__version__
+            except AttributeError:
+                # Module has no __version__ attribute
+                pass
+            except Exception as x:
+                # Unable to access module
+                logging.error(f"{x} when attempting to access {mod} version using .__version__")
+                pass
+            else:
+                continue
+
+            try:
+                module_versions_dict[mod] = __import__(mod).version
+            except AttributeError:
+                # Module has no .version attribute
+                pass
+            except Exception as x:
+                # Unable to access module
+                logging.error(f"{x} when attempting to access {mod} version using .version")
+                pass
+            else:
+                continue
+
+            # Unreliable, so last option
+            try:
+                module_versions_dict[mod] = pkg_resources.get_distribution(mod).version
+            except:
+                # Modules that cannot be found by pkg_resources
+                pass
+            else:
+                continue
+
+            # Below is code that calculates the version of each module using pip, however it is
+            # very time consuming, and only is useful for a handful of modules
+            # try:
+            #     pip_module_show = str(subprocess.check_output(f"pip show {mod}"), 'utf-8')
+            #     show_list = pip_module_show.replace("\r\n", ",").split(",")
+            #     for sec in show_list:
+            #         if sec.startswith("Version"):
+            #             module_versions_dict[mod] = sec.split(":")[1].strip()
+            #         else:
+            #             # Unalbe to get  version for this specific module
+            #             pass
+            # except Exception as x:
+            #     # Module not available through pip
+            #     logging.error(f"{x} when attempting to get the version of {mod} through pip")
+            #     pass
+            # else:
+            #     continue
+
+            # Modules version number could not be attained by any of the previous methods
+            unattainable_modules.append(mod)
+
+        # Clean up
+        module_versions_dict = self.remove_duplicate_modules(module_versions_dict)
+        unattainable_modules = self.format_unattainable_modules_list(module_versions_dict, unattainable_modules)
```

Comment:
Don't know why you would want to list unknown packages separately. My suggestions above produce `package_name: "unknown"` or `package_name: "unknown: import exception"`, which seems like it will be more useful.

Reply:
Added